pkg/api/validation/validation.go

```diff
@@ -2272,6 +2276,20 @@ func ValidatePodSpec(spec *api.PodSpec, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, ValidateHostAliases(spec.HostAliases, fldPath.Child("hostAliases"))...)
 	}
 
+	if len(spec.PriorityClassName) > 0 {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.PodPriority) {
```

Comment:
in my version of the code, the feature gate is not checked when priority class is not specified. Also, the condition "if len(spec.PriorityClassName) > 0" should be repeated in both if and else blocks of your version.